File: Counting/FrogRiverOne.js
/**
 * Created by muhammadmuhaimin on 2015-10-25.
 */
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');


function solution(X, A) {
  // write your code in JavaScript (Node.js 4.0.0)
  var elementCount = [];
  var steps = X;
  for (var i = 0; i < X + 1; i++) {
    elementCount.push(0);
  }
  for (i = 0; i < A.length; i++) {
    if (A[i] <= X && elementCount[A[i]] === 0) {
      elementCount[A[i]] = 1;
      steps--;
    }
    if (steps === 0) {
      return i;
    }
  }
  return -1;
}

File: Counting/FrogRiverOne.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, A):
    # write your code in Python 3.6
    fallen_leaf_count = [0] * X
    total_leaf = 0
    for i in range (0, len(A)):
        if fallen_leaf_count[A[i] - 1] == 0:
            total_leaf += 1
        fallen_leaf_count[A[i] - 1] = 1
        if total_leaf == X:
            return i
    return -1
